loopy/transform/callable.py

```diff
@@ -154,6 +154,81 @@ def __call__(self, target, identifier):
         return None
 
 
+def subarrayrefs_are_equiv(sar1, sar2, knl):
+    """
+    Compares if two instance of :class:`loopy.symbolic.SubArrayRef`s point
+    to the same array region.
+    """
+    from loopy.kernel.function_interface import get_arg_descriptor_for_expression
+
+    return get_arg_descriptor_for_expression(knl, sar1) == (
+            get_arg_descriptor_for_expression(knl, sar2)) and (
+                    sar1.get_begin_subscript(knl) ==
+                    sar2.get_begin_subscript(knl))
+
+
+def _check_correctness_of_args_and_assignees(insn, callee_kernel, caller_knl):
+    from loopy.kernel.function_interface import get_kw_pos_association
+    kw_to_pos, pos_to_kw = get_kw_pos_association(callee_kernel)
+    callee_args_to_insn_params = [[] for _ in callee_kernel.args]
+    expr = insn.expression
+    from pymbolic.primitives import Call, CallWithKwargs
+    if isinstance(expr, Call):
+        expr = CallWithKwargs(expr.function, expr.parameters, kw_parameters={})
+    for i, param in enumerate(expr.parameters):
+        pos = kw_to_pos[callee_kernel.args[i].name]
+        if pos < 0:
+            raise LoopyError("#{} argument meant for output obtained as an"
+                    " input in '{}'.".format(i, insn))
```

Comment:
- [ ] Use 1-based index.
- [ ] Clarify message to state that.

Reply:
Done in 6f177eb9.